File: src/main/java/net/softbell/bsh/iot/controller/rest/v1/IotActionRestV1.java
package net.softbell.bsh.iot.controller.rest.v1;

import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import lombok.AllArgsConstructor;
import net.softbell.bsh.dto.response.ResultDto;
import net.softbell.bsh.iot.service.v1.IotNodeServiceV1;
import net.softbell.bsh.service.ResponseService;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : IoT 액션 REST API 컨트롤러 V1
 */
@AllArgsConstructor
@RestController
@RequestMapping("/api/rest/v1/iot/action")
public class IotActionRestV1
{
    private final ResponseService responseService;
    private final IotNodeServiceV1 iotNodeService;
	
	@PostMapping("/exec/{id}")
	public ResultDto setNodeItemValue(@PathVariable("id")long actionId)
	{
		// Field
		boolean isSuccess;
		
		// Init
		isSuccess = false; //iotNodeService.setItemValue(id, value); // TODO
		
		// Return
		if (isSuccess)
			return responseService.getSuccessResult();
		else
			return responseService.getFailResult(-10, "해당하는 아이템이 없음");
	}
}

File: src/main/kotlin/net/softbell/bsh/iot/controller/rest/v1/IotActionRestV1.kt
package net.softbell.bsh.iot.controller.rest.v1

import net.softbell.bsh.dto.response.ResultDto
import net.softbell.bsh.iot.service.v1.IotActionServiceV1
import net.softbell.bsh.service.ResponseService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : IoT 액션 REST API 컨트롤러 V1
 */
@RestController
@RequestMapping("/api/rest/v1/iot/action")
class IotActionRestV1 {
    @Autowired lateinit var responseService: ResponseService
    @Autowired lateinit var iotActionService: IotActionServiceV1

    @PostMapping("/exec/{id}")
    fun execNodeAction(auth: Authentication, @PathVariable("id") actionId: Long): ResultDto? {
        // Field
        val isSuccess: Boolean

        // Init
        isSuccess = iotActionService!!.execAction(actionId, auth)

        // Return
        return if (isSuccess) responseService.getSuccessResult() else responseService!!.getFailResult(-10, "해당하는 아이템이 없음")
    }
}